lib/http.js

```diff
@@ -6,9 +6,43 @@ const EventEmitter = require('events').EventEmitter;
 
 exports.IncomingMessage = require('_http_incoming').IncomingMessage;
 
-
-const common = require('_http_common');
-exports.METHODS = common.methods.slice().sort();
+// Request Methods
+var METHODS = [
+  'DELETE',
+  'GET',
+  'HEAD',
+  'POST',
+  'PUT',
+  // pathological
+  'CONNECT',
+  'OPTIONS',
+  'TRACE',
+  // webdav
+  'COPY',
+  'LOCK',
+  'MKCOL',
+  'MOVE',
+  'PROPFIND',
+  'PROPPATCH',
+  'SEARCH',
+  'UNLOCK',
+  // subversion
+  'REPORT',
+  'MKACTIVITY',
+  'CHECKOUT',
+  'MERGE',
+  // upnp
+  'MSEARCH',
+  'NOTIFY',
+  'SUBSCRIBE',
+  'UNSUBSCRIBE',
+  // RFC-5789
+  'PATCH',
+  'PURGE'
+];
+exports.__defineGetter__('METHODS', util.deprecate(function() {
```

Comment:
I dunno, I just kept the list there for easy backwards compatibility, as the js parser allows any syntactically valid method. I just copied the `__defineGetter__` from somewhere else that was deprecating property access.